acceptance/lib/puppet/acceptance/temp_file_utils.rb

```diff
@@ -140,17 +140,13 @@ def chmod(host, mode, path)
       # value containing only the file mode, excluding the type, e.g
       # S_IFDIR 0040000
       def stat(host, path)
-        stat_command = case host['platform']
-                       when /osx/
-                         "stat -f '%Su:%Sg:%p'"
-                       else
-                         "stat --format '%U:%G:%a'"
-                       end
+        require File.join(File.dirname(__FILE__),'common_utils.rb')
+        ruby = Puppet::Acceptance::CommandUtils.ruby_command(host)
+        owner = on(host,"ls -l -d #{path} | awk '{print $3;}'").stdout.chomp
+        group = on(host,"ls -l -d #{path} | awk '{print $4;}'").stdout.chomp
+        mode = on(host, "#{ruby} -e 'puts (File.stat(\"#{path}\").mode & 07777)'").stdout.chomp.to_i
```

Comment:
since we're calling File.stat to get mode, you could have it return owner and group as well